processor/batchprocessor/batch_processor.go

```diff
@@ -206,9 +210,12 @@ func (bp *batchProcessor) sendItems(measure *stats.Int64Measure) {
 		_ = stats.RecordWithTags(context.Background(), statsTags, statBatchSendSizeBytes.M(int64(bp.batch.size())))
 	}
 
-	if err := bp.batch.export(context.Background()); err != nil {
-		bp.logger.Warn("Sender failed", zap.Error(err))
-	}
+	go func() {
```

Comment:
In general, I prefer to use queues/workers, so that we don't risk having tons of go funcs at the same time without any restriction. Do you think it would be easy and feasible to make one here?

Reply:
Generally, I totally agree with workers pool. But currently here I see more problems with them. 
Firstly, what are fallbacks? Imagine collector is overloaded and workers aren't able to do their job quickly. What would we do? Block until worker returns? Then let me remind that it is the only worker in batch processor, so this blocks he whole processor. 
Secondly, there are still no limits for receivers' consumes. 
Thirdly, optimal number of workers might be changed with changes in pipeline. For example, you have 4 workers, which deliver traces from batch processor directly to another async component (some exporter with queue). Then actually these workers do no work. Then you introduce a new component in between, which processes each trace in a second (io block, not cpu-bound). Then we have a max throughput 4 traces/s. Very implicit behaviour. 
At last, batch processor (always?) has a time interval between exporting. This won't be the thing producing 100+ goroutines/s actually

Reply:
>Imagine collector is overloaded and workers aren't able to do their job quickly. 

That's precisely why worker queues are used so that we don't create goroutines indefinitely, overloading even more what might already be an overloaded system.

Reply:
Well, address my points in the comment above and I'd be happy to add pool

Reply:
I agree with @jpkrohling that there needs to be a configurable upper limit on the number of concurrent sends from the batch processor. For pipelines where the batch processor is sending directly to an async processor, I don't think think it's reasonable to default to an infinite number of goroutines that can be created.

For pipelines that are immediately followed by a exporter that uses a queued retry sending (eg. otlpexporter), creating a new goroutine for each send isn't very useful and should be configurable to set the parallelization to 1.

Reply:
Okay, an optional configurable upper limit seems acceptable. If someone wants to possibly shoot his leg switching from async exporting to sync, let it be.

Any thoughts about the default value? No limit or 1? (NumProcs isn't considered, because it doesn't tolerate cgroup limits and looks at physical cores number. Not really great for docker/kubertenes/...)

Reply:
1 would be similar to the current behavior, right? That would be my choice.